petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -29,19 +28,37 @@ def test_primitive(self):
         ])
         df = self.spark.createDataFrame([
             (True, 0.12, 432.1, 5, 5, 0),
-            (False, 123.45, 0.987, 9, 908, 765)],
-            schema=schema)
+            (False, 123.45, 0.987, 9, 908, 765)], schema=schema).coalesce(1)
+        # If we use numPartition > 1, the order of the loaded dataset would be non-deterministic.
+        expected_df = df.collect()
 
         converter = make_spark_converter(df)
         with converter.make_tf_dataset() as dataset:
             iterator = dataset.make_one_shot_iterator()
             tensor = iterator.get_next()
             with tf.Session() as sess:
                 ts = sess.run(tensor)
+            for i in range(converter.dataset_size):
+                for col in df.schema.names:
+                    self.assertEquals(getattr(ts, col)[i], expected_df[i][col])
+
+            self.assertEquals(len(converter), len(expected_df))
+
```

Comment:
There is one issues in this test:
You only check one batch read from petastorm reader, current implementation one batch is one row group in parquet(note the batch size is variable in each row group) ,here your test dataframe is small enough so that there're only one group.
So Mark a TODO here to say: we will improve the test once the batch_size argument added.

Reply:
sure

Reply:
Added at L41